src/fx/Storage.js

```diff
@@ -0,0 +1,102 @@
+import { assign } from "../utils.js"
+
+function writeToStorage(props) {
+  var storage = window[(props.storage || "session") + "Storage"]
+  var value = props.converter(props.value)
+  storage.setItem(props.key, value)
+}
+
+function readFromStorage(props, dispatch) {
+  var storage = window[(props.storage || "session") + "Storage"]
+  var value = props.converter(storage.getItem(props.key))
+  var dispatchProps = assign({}, props.props || {})
+  dispatchProps[props.prop || "value"] = value
+  dispatch(props.action, dispatchProps)
+}
+
+function removeFromStorage(props) {
+  var storage = window[(props.storage || "session") + "Storage"]
+  storage.removeItem(props.key)
+}
+
+/**
+ * Describes an effect that will write a key value pair to Storage. By default the item is written to `sessionStorage`, to write to `localStorage` set the `storage` prop to `local`. Values can be saved as json by specifying `json` as true.
+ *
+ * @memberof module:fx
+ * @param {object} props
+ * @param {string} props.key - Specify key to use
+ * @param {*} props.value - Value to write to storage
+ * @param {string} props.storage - Storage area to write to, can be either "session" or "local"
+ * @param {boolean} props.json - Converts the value of the item to JSON
+ * @param {function} props.converter - Use a custom converter function to encode the value of the item
```

Comment:
Do we need to support custom converters or just provide JSON and leave any conversion up to the caller?

Reply:
We could use a JSON converter by default and allow the caller to override the converter. This would eliminate the need for `props.json`

Reply:
I think JSON is so commonly used that this is probably the right way to go. Similar to how the `Http` effect works.